e2e/tcproute_test.go

```diff
@@ -334,18 +331,22 @@ func TestRgraphTCPRouteAddBackends(t *testing.T) {
 		t.Fatalf("buildTCPRoute(_, tcproute-test, _, _, _) = (_, %v), want (_, nil)", err)
 	}
 	t.Logf("TCPRoute created: %v", tcprID)
-	t.Cleanup(func() {
-		err := theCloud.TcpRoutes().Delete(ctx, tcprID.Key)
-		t.Logf("theCloud.TcpRoutes().Delete(_, %s): %v", tcprID.Key, err)
-	})
 
 	expectedActions := []exec.ActionMetadata{
 		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, tcprID)},
 		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, bsID)},
 		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, hcID)},
 		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, negID)},
 	}
+
 	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+	t.Cleanup(func() {
+		var allErros error
+		allErros = errors.Join(allErros, theCloud.TcpRoutes().Delete(ctx, tcprID.Key))
+		t.Logf("theCloud.TcpRoutes().Delete(_, %s): %v", tcprID.Key, err)
```

Comment:
why do you need all errors if you print err? 
You should delete hc and negs here also

Reply:
Done.